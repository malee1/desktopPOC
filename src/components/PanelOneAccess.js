import { React, useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import {
  Accordion,
  AccordionSummary,
  AccordionDetails,
  Box,
  Typography,
  Collapse,
  Button
} from '@material-ui/core';
import { ArrowDropDown, ExpandMore } from '@material-ui/icons';
import { useSelector } from 'react-redux';
import { panelOneLabels } from '../constants/panelLabels';
import DropDownPanel from './DropDownPanel';

const useStyles = makeStyles({
  root: {
    flexGrow: 1,
    padding: '5px'
  },
  box: {
    width: '27vw'
  },
  list: {
    flexDirection: 'column',
    whiteSpace: 'nowrap'
  },
  listItem: {
    backgroundColor: 'white'
  },
  listItemButton: {
    justifyContent: 'flex-end',
    backgroundColor: 'white',
    display: 'flex'
  },
  iconContainer: {
    width: '8%',
    backgroundColor: 'white'
  },
  icon: {
    backgroundColor: 'white'
  },
  labelContainer: {
    width: '100%',
    marginLeft: '2px'
  },
  label: {
    backgroundColor: 'white',
    paddingLeft: '5px'
  },
  labelValueContainer: {
    width: '55%',
    marginLeft: '2px'
  },
  labelValue: {
    backgroundColor: 'white',
    paddingLeft: '5px'
  },
  dropDown: {
    // backgroundColor: 'grey',
    zIndex: '1000',
    position: 'absolute',
    height: '15vh',
    width: '27vw'
  },
  panelButton: {
    height: '30px'
  },
  titleContainer: {
    display: 'flex'
  },
  accordianDetails: {
    zIndex: '1500',
    // position: 'absolute',
    backgroundColor: 'white',
    width: '100%'
  }
});

export default function PanelOne() {
  const classes = useStyles();
  const labels = panelOneLabels;

  const [open, setOpen] = useState(false);

  const handleDropDown = () => {
    setOpen(!open);
  };

  const customer = useSelector((state) => state.customer.data);

  return (
    <div className={classes.root} data-testid="panel-one-access">
      <Box className={classes.box}>
        <Accordion data-testid="accordian">
          <AccordionSummary className={classes.accordion}>
            <div className={classes.labelContainer}>
              <Typography className={classes.label} variant="h5" data-testid="aaa">
                {labels.name} {customer.name}
              </Typography>
            </div>
          </AccordionSummary>
        </Accordion>
        <Accordion data-testid="accordian">
          <AccordionSummary
            expandIcon={<ExpandMore />}
            aria-controls="panel1a-content"
            id="panel1a-header"
            className={classes.accordion}
          >
            <div className={classes.labelContainer}>
              <Typography className={classes.label} variant="h5" data-testid="aaa">
                {labels.dob}
              </Typography>
            </div>
          </AccordionSummary>
          <AccordionDetails className={classes.accordianDetails}>
            <div className={classes.labelValueContainer}>
              <Typography className={classes.labelValue} variant="h4">
                {customer.dob}
              </Typography>
            </div>
          </AccordionDetails>
        </Accordion>
        <Accordion data-testid="accordian">
          <AccordionSummary
            expandIcon={<ExpandMore />}
            aria-controls="panel1a-content"
            id="panel1a-header"
            className={classes.accordion}
          >
            <div className={classes.labelContainer}>
              <Typography className={classes.label} variant="h5" data-testid="aaa">
                {labels.uRef}
              </Typography>
            </div>
          </AccordionSummary>
          <AccordionDetails className={classes.accordianDetails}>
            <div className={classes.labelValueContainer}>
              <Typography className={classes.labelValue} variant="h4">
                {customer.uRef}
              </Typography>
            </div>
          </AccordionDetails>
        </Accordion>
        <Accordion data-testid="accordian">
          <AccordionSummary
            expandIcon={<ExpandMore />}
            aria-controls="panel1a-content"
            id="panel1a-header"
            className={classes.accordion}
          >
            <div className={classes.labelContainer}>
              <Typography className={classes.label} variant="h5" data-testid="aaa">
                {labels.tel}
              </Typography>
            </div>
          </AccordionSummary>
          <AccordionDetails className={classes.accordianDetails}>
            <div className={classes.labelValueContainer}>
              <Typography className={classes.labelValue} variant="h4">
                {customer.tel}
              </Typography>
            </div>
          </AccordionDetails>
        </Accordion>
        <Button
          className={classes.panelButton}
          onClick={handleDropDown}
          variant="contained"
          color="secondary"
          data-testid="button"
        >
          <Typography className={classes.buttonLabel} variant="body2">
            More Info
          </Typography>
          <ArrowDropDown className={classes.dropDownIcon} />
        </Button>

        <Collapse in={open} timeout="auto" unmountOnExit>
          <div className={classes.dropDown}>
            <DropDownPanel />
          </div>
        </Collapse>
      </Box>
    </div>
  );
}
