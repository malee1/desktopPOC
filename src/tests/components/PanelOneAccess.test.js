import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { cleanup, render, fireEvent } from '@testing-library/react';

import PanelOneAccess from '../../components/PanelOneAccess';

const mockStore = configureMockStore();
const store = mockStore({
  customer: {
    data: {
      name: 'Test Test',
      dob: '1111',
      uRef: '2222',
      tel: '3333',
      timeWait: '10 mins',
      context: 'test test',
      recording: 'Inactive',
      callInfo: 'ABCDE',
      alerts: 'Important Info 1',
      service: 'Alert1',
      keyInfo: 'Info 1',
      auth: 'Approved'
    }
  }
});

afterEach(() => {
  cleanup();
  jest.clearAllMocks();
});

describe('the panel renders correctly', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <PanelOneAccess />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has 4 accordian sections', () => {
    const { getAllByTestId } = render(
      <Provider store={store}>
        <PanelOneAccess />
      </Provider>
    );
    const container = getAllByTestId('accordian');
    expect(container.length).toBe(4);
  });
});

describe('button click works', () => {
  it('on click the function is called', () => {
    const { getByText, getAllByTestId } = render(
      <Provider store={store}>
        <PanelOneAccess />
      </Provider>
    );
    const btn = getByText('More Info');
    fireEvent.click(btn);
    const dropPanel = getAllByTestId('drop-down-panel');
    expect(dropPanel.length).toBe(1);
  });
});
