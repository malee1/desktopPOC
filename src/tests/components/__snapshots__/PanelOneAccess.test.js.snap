// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the panel renders correctly matches the snapshot 1`] = `
<div
  className="makeStyles-root-1"
  data-testid="panel-one-access"
>
  <div
    className="MuiBox-root MuiBox-root-16 makeStyles-box-2"
  >
    <div
      className="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
      data-testid="accordian"
    >
      <div
        aria-disabled={false}
        aria-expanded={false}
        className="MuiButtonBase-root MuiAccordionSummary-root"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiAccordionSummary-content"
        >
          <div
            className="makeStyles-labelContainer-8"
          >
            <h5
              className="MuiTypography-root makeStyles-label-9 MuiTypography-h5"
              data-testid="aaa"
            >
              Name: 
               
              Test Test
            </h5>
          </div>
        </div>
      </div>
      <div
        className="MuiCollapse-container MuiCollapse-hidden"
        style={
          Object {
            "minHeight": "0px",
          }
        }
      >
        <div
          className="MuiCollapse-wrapper"
        >
          <div
            className="MuiCollapse-wrapperInner"
          >
            <div
              role="region"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
      data-testid="accordian"
    >
      <div
        aria-controls="panel1a-content"
        aria-disabled={false}
        aria-expanded={false}
        className="MuiButtonBase-root MuiAccordionSummary-root"
        id="panel1a-header"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiAccordionSummary-content"
        >
          <div
            className="makeStyles-labelContainer-8"
          >
            <h5
              className="MuiTypography-root makeStyles-label-9 MuiTypography-h5"
              data-testid="aaa"
            >
              Date of Birth: 
            </h5>
          </div>
        </div>
        <div
          aria-disabled={false}
          aria-hidden={true}
          className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role={null}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        className="MuiCollapse-container MuiCollapse-hidden"
        style={
          Object {
            "minHeight": "0px",
          }
        }
      >
        <div
          className="MuiCollapse-wrapper"
        >
          <div
            className="MuiCollapse-wrapperInner"
          >
            <div
              aria-labelledby="panel1a-header"
              id="panel1a-content"
              role="region"
            >
              <div
                className="MuiAccordionDetails-root makeStyles-accordianDetails-15"
              >
                <div
                  className="makeStyles-labelValueContainer-10"
                >
                  <h4
                    className="MuiTypography-root makeStyles-labelValue-11 MuiTypography-h4"
                  >
                    1111
                  </h4>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
      data-testid="accordian"
    >
      <div
        aria-controls="panel1a-content"
        aria-disabled={false}
        aria-expanded={false}
        className="MuiButtonBase-root MuiAccordionSummary-root"
        id="panel1a-header"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiAccordionSummary-content"
        >
          <div
            className="makeStyles-labelContainer-8"
          >
            <h5
              className="MuiTypography-root makeStyles-label-9 MuiTypography-h5"
              data-testid="aaa"
            >
              Unique Reference: 
            </h5>
          </div>
        </div>
        <div
          aria-disabled={false}
          aria-hidden={true}
          className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role={null}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        className="MuiCollapse-container MuiCollapse-hidden"
        style={
          Object {
            "minHeight": "0px",
          }
        }
      >
        <div
          className="MuiCollapse-wrapper"
        >
          <div
            className="MuiCollapse-wrapperInner"
          >
            <div
              aria-labelledby="panel1a-header"
              id="panel1a-content"
              role="region"
            >
              <div
                className="MuiAccordionDetails-root makeStyles-accordianDetails-15"
              >
                <div
                  className="makeStyles-labelValueContainer-10"
                >
                  <h4
                    className="MuiTypography-root makeStyles-labelValue-11 MuiTypography-h4"
                  >
                    2222
                  </h4>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
      data-testid="accordian"
    >
      <div
        aria-controls="panel1a-content"
        aria-disabled={false}
        aria-expanded={false}
        className="MuiButtonBase-root MuiAccordionSummary-root"
        id="panel1a-header"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiAccordionSummary-content"
        >
          <div
            className="makeStyles-labelContainer-8"
          >
            <h5
              className="MuiTypography-root makeStyles-label-9 MuiTypography-h5"
              data-testid="aaa"
            >
              Tel: 
            </h5>
          </div>
        </div>
        <div
          aria-disabled={false}
          aria-hidden={true}
          className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role={null}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        className="MuiCollapse-container MuiCollapse-hidden"
        style={
          Object {
            "minHeight": "0px",
          }
        }
      >
        <div
          className="MuiCollapse-wrapper"
        >
          <div
            className="MuiCollapse-wrapperInner"
          >
            <div
              aria-labelledby="panel1a-header"
              id="panel1a-content"
              role="region"
            >
              <div
                className="MuiAccordionDetails-root makeStyles-accordianDetails-15"
              >
                <div
                  className="makeStyles-labelValueContainer-10"
                >
                  <h4
                    className="MuiTypography-root makeStyles-labelValue-11 MuiTypography-h4"
                  >
                    3333
                  </h4>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-panelButton-13 MuiButton-containedSecondary"
      data-testid="button"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <p
          className="MuiTypography-root MuiTypography-body2"
        >
          More Info
        </p>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
